spec/main.tex

```diff
@@ -1100,158 +1809,203 @@ \subsubsection{Non-Threshold ($n$-out-of-$n$) Presigning} \label{section:orig-pr
-        \item If shares are additive\footnote{In this case we have $t=n$.} shares of the private key, 
-            set $x_{i} = x'_{S(i)}$, $\vec X = (X'_{S(j)})_{j \in [n]}$.
-        
-        \item If shares are Shamir secret shares of the private key: 
+    \item[\bf Input.] \
+    \begin{itemize}[noitemsep,topsep=0pt]
+        \item size of signing set $t \ge 2$,
+        \item party index $i \in [t]$,
+        \item injective index map $S : [t] \to [n]$ that maps signers indicies at signing to their indicies at key generation, i.e. $S(i)$ is the index that $P_i$ had at key-generation time,
+        \item key share $K_{S(i)}$,
+        \item context separation string $\sid \in \Byte^*$,
+        \item additive shift $\var{shift} \in \Z_q$ for HD child derivation. $\var{shift} = 0$ disables HD derivation. Deriving additive shift is up to specific standard of HD-wallets, e.g. see \cite{bip32} or \cite{slip10} \label{deriving-shift-note}
+        \item security level $L$ (see Section~\ref{sec:security-params}),
+        \item flag $\var{echo\_enabled} \in \Bit$ indicating whether reliability check is enabled,
+        \item curve $\E$ with generator $G$ of prime order $q$
+    \end{itemize}
+
+    \item[\bf Map threshold signing protocol to non-threshold by transforming the shares] \
+    \begin{algorithmic}[1]
+        \State Extract data from key share $K_{S(i)}$, which contains:
+
         \begin{itemize}
-            \item For $j \in [t]$, compute Lagrange coefficient $\lambda_j = \prod\limits_{m \in [t]\setminus\{j\}} \frac{I(S(m))}{I(S(m)) - I(S(j))} \bmod{q}$.
-            \item Compute $x_{i} = \lambda_i \cdot x'_{S(i)}$.
-            \item For $j \in [n]$, compute $X_j= \lambda_j \cdot X'_{S(j)}$; then set $\vec X = \{X_j\}_{j \in [t]}$.
+        \item threshold value $\threshold$,
+        \item number of key holders~$n$,
+        \item secret share $x'_{S(i)} \in \Z_q$,
+        \item parties' public shares $\vec X' = (X'_j)_{j \in [n]} \in \E^n$,
+        \item an injective map $I : [n] \to \Fq \setminus \{0\}$,
+        \item Paillier secret key $\sk'_{S(i)}$,
+        \item parties' Paillier keys $\vec N' = (N'_j)_{j \in [n]} \in \Z^n$,
+        \item parties' auxiliary information $\vec R' = (s_j, t_j, \hat N)_{j \in [n]} \in (\Z, \Z, \Z)^n$
         \end{itemize}
-    \end{itemize}
 
-    \item[\bf Step 2.] {\bf (HD-wallets)} If HD-wallets support enabled:
-    \begin{itemize}
-      \item Set $X_0 := X_0 + \text{shift} \cdot G$
-      \item If $i = 0$, set $x_0 := x_0 + \text{shift}$
-      \item Note: output signature will be valid for public key $Y + \text{shift} \cdot G$
-    \end{itemize}
+        \State Set $\sk_i = \sk'_{S(i)}$ and $\vec R = (R'_{S(j)})_{j \in [n]}$
+        \If{shares are additive\footnote{In this case we have $t=n$.} shares of the private key}
+            \State Set $x_{i} = x'_{S(i)}$, $\vec X = (X'_{S(j)})_{j \in [n]}$
+        \EndIf
+        \If{shares are Shamir secret shares of the private key}
```

Comment:
I think this should rather be `\elseif`, does algorithmic support it?

Reply:
I tried exactly this, `\Elseif`, it gave me an error, I didn't want to mess with it, so I made a bunch of `\If`